src/modules/Elsa.MassTransit/Services/MassTransitWorkflowDispatcher.cs

```diff
@@ -74,31 +77,97 @@ await sendEndpoint.Send(new DispatchWorkflowInstance(request.InstanceId)
+            {
+                Workflow = workflow,
+                WorkflowInstanceId = request.WorkflowInstanceId,
+                Input = request.Input,
+                Properties = request.Properties,
+                CorrelationId = request.CorrelationId
+            };
+
+            // The workflow instance is created and persisted in the database.
+            var workflowInstance = await workflowInstanceManager.CreateWorkflowInstanceAsync(createWorkflowInstanceRequest, cancellationToken);
+
+            // The workflow instance is then dispatched for execution.
+            var sendEndpoint = await GetSendEndpointAsync(options);
+            var message = DispatchWorkflowDefinition.DispatchExistingWorkflowInstance(workflowInstance.Id, trigger.ActivityId);
+            await sendEndpoint.Send(message, cancellationToken);
+        }
+    }
+
+    private async Task DispatchBookmarksAsync(DispatchTriggerWorkflowsRequest request, DispatchWorkflowOptions? options = default, CancellationToken cancellationToken = default)
+    {
+        var correlationId = request.CorrelationId;
+        var workflowInstanceId = request.WorkflowInstanceId;
+        var activityInstanceId = request.ActivityInstanceId;
+        var bookmarkHash = bookmarkHasher.Hash(request.ActivityTypeName, request.BookmarkPayload, activityInstanceId);
+
+        var filter = new BookmarkFilter { Hash = bookmarkHash, CorrelationId = correlationId, WorkflowInstanceId = workflowInstanceId, ActivityInstanceId = activityInstanceId };
+        var bookmarks = (await bookmarkStore.FindManyAsync(filter, cancellationToken)).ToList();
+
+        await DispatchBookmarksAsync(bookmarks, request.Input, request.Properties, options, cancellationToken);
+    }
+
+    private async Task DispatchBookmarksAsync(IEnumerable<StoredBookmark> bookmarks, IDictionary<string, object>? input, IDictionary<string, object>? properties, DispatchWorkflowOptions? options, CancellationToken cancellationToken)
+    {
+        foreach (var bookmark in bookmarks)
+        {
+            if (input != null || properties != null)
+            {
+                var workflowInstance = await workflowInstanceManager.FindByIdAsync(bookmark.WorkflowInstanceId, cancellationToken);
+
+                if (workflowInstance == null)
+                {
+                    logger.LogWarning("Workflow instance with ID '{WorkflowInstanceId}' not found", bookmark.WorkflowInstanceId);
+                    continue;
+                }
+
+                if (input != null) workflowInstance.WorkflowState.Input.Merge(input);
+                if (properties != null) workflowInstance.WorkflowState.Properties.Merge(properties);
+
+                await workflowInstanceManager.SaveAsync(workflowInstance, cancellationToken);
+            }
+
+            var dispatchInstanceRequest = new DispatchWorkflowInstanceRequest(bookmark.BookmarkId) { BookmarkId = bookmark.BookmarkId, CorrelationId = bookmark.CorrelationId };
```

Comment:
```suggestion
           var workflow = await workflowDefinitionService.FindWorkflowAsync(trigger.WorkflowDefinitionVersionId, cancellationToken);

            if (workflow == null)
            {
                logger.LogWarning("Workflow definition with ID '{WorkflowDefinitionId}' not found", trigger.WorkflowDefinitionVersionId);
                continue;
            }

            var createWorkflowInstanceRequest = new CreateWorkflowInstanceRequest
            {
                Workflow = workflow,
                WorkflowInstanceId = request.WorkflowInstanceId,
                Input = request.Input,
                Properties = request.Properties,
                CorrelationId = request.CorrelationId
            };

            // The workflow instance is created and persisted in the database.
            var workflowInstance = await workflowInstanceManager.CreateWorkflowInstanceAsync(createWorkflowInstanceRequest, cancellationToken);

            // The workflow instance is then dispatched for execution.
            var sendEndpoint = await GetSendEndpointAsync(options);
            var message = DispatchWorkflowDefinition.DispatchExistingWorkflowInstance(workflowInstance.Id, trigger.ActivityId);
```

Reply:
This suggestions fixes sending the workflowInstanceId instead of the bookmarkId as the instanceId. Preventing resuming the workflows.

Reply:
That's a really good catch!